ui/src/main/resources/edu/wpi/grip/ui/templates/java/Switch.vm

```diff
@@ -0,0 +1,50 @@
+#set($needsMut = true)
+		//input
+#set($boolInp = $step.getInput(0))
+#if( $boolInp.hasValue() )##Has a set value
+		$boolInp.baseType() $boolInp.name()$c = $boolInp.value();
+#else##Has Connected Socket
+		$boolInp.baseType() $boolInp.name()$c = ${boolInp.value()}.get().$boolInp.baseType()Value();
+#end
+#set( $inputTrue = $step.getInput(1) )
+#if( $inputTrue.mutable() )##Is Mutable
+#if( $inputTrue.number() )##Is Number
+		Number $inputTrue.name()$c = ${inputTrue.value()}.get();
+#else##Not number
+		$inputTrue.type() $inputTrue.name()$c = ${inputTrue.value()}.get();
+#end##end Is Number
+#else##Not Mutable
+#if ( $inputTrue.type() == "FlipCode" )
+		$inputTrue.type() $inputTrue.name()$c = FlipCode.$inputTrue.value();
+#else
+		$inputTrue.type() $inputTrue.name()$c = $inputTrue.value();
+#end
+#end##end Is Mutable
+#set( $inputFalse = $step.getInput(2) )
+#if( $inputFalse.mutable() )##Is Mutable
+#if( $inputFalse.number() )##Is Number
+		Number $inputFalse.name()$c = ${inputFalse.value()}.get();
+#else##Not number
+		$inputFalse.type() $inputFalse.name()$c = ${inputFalse.value()}.get();
+#end##end Is Number
+#else##Not Mutable
+#if ( $inputFalse.type() == "FlipCode" )
+		$inputFalse.type() $inputFalse.name()$c = FlipCode.$inputFalse.value();
+#else
+		$inputFalse.type() $inputFalse.name()$c = $inputFalse.value();
+#end
+#end##end Is Mutable
+#set( $out = $step.getOutput(0) )
+#set ( $outMutName = $out.name() )
+#set ( $outMutName = $outMutName + "Mutate" )
+#set ( $outMutName = $outMutName + $c )
+		//output
+		#MutOf($out.baseType()) $outMutName = new #MutOf($out.baseType())();
+		pipelineSwitch($boolInp.name()$c, $inputTrue.name()$c, $inputFalse.name()$c, $outMutName);
+		//output assignment
+#if ($out.mutable())
+		#MutOf($out.type()) $out.name() = $outMutName;
+#else
+		$out.type() $out.name() = ${outMutName}.get();
+#end
+		outputs.put("$out.name()", $out.name());
```

Comment:
This file is complicated. I bet that with judicious application of macros, it could be compressed down to something like:

```
#set( $out = $step.getOutput(0) )
#set ( $outMutName = $out.name() )
#set ( $outMutName = $outMutName + "Mutate" )
#set ( $outMutName = $outMutName + $c )

		#MutOf($out.baseType())  $outMutName = new #MutOf($out.baseType())();

		pipelineSwitch( #InputAccessExpression( $boolInp ), #InputAccessExpression( $inputTrue ), #InputAccessExpression( $inputFalse), $outMutName);

#if ($out.mutable())
		#MutOf($out.type()) $out.name() = $outMutName;
#else
		$out.type() $out.name() = ${outMutName}.get();
#end
		outputs.put("$out.name()", $out.name());
```